doc/cprover-manual/smt2-incr.md

```diff
@@ -0,0 +1,40 @@
+[CPROVER Manual TOC](../)
+
+## Incremental SMT backend
+
+CBMC comes with a solver-agnostic incremental SMT backend.
+
+The SMT incremental backend supports the following `C` primitives:
+
+- Bit vector arithmetics
+- Pointers
+- Arrays (support is currently incomplete)
+
+Primitive not supported are:
+
+- Floating point values and arithmetics
+- Structs
+
+Requirement to use the incremental SMT backend is to have a solver which accepts
+SMTlib 2.6 incremental input from the standard input and that supports the
+`QF_AUFBV` (Quantifier Free Array Uninterpreted Functions and Bit Vectors).
+
+To use this functionality it is enough to add the argument
+`--incremental-smt2-solver <cmd>` where `<cmd>` is the command to invoke the SMT
+solver of choice using the interactive mode and accepting input from the
+standard input.
+
+Examples are:
+
+1. `--incremental-smt2-solver 'z3 -smt2 -in'` to use the Z3 solver.
+2. `--incremental-smt2-solver 'cvc5 --lang=smtlib2.6 --incremental'` to use the
+   CVC5 solver.
+
+Note that the solver has to be user-supplied and CBMC does not come packaged
+with any SMT solver, so the solver name must be in the `PATH` or an executable
+with full path must be provided.
+
+Also note that because of the partial array support, the `--slice-formula`
```

Comment:
I would phrase that as

> Due to lack of support for conversion of `array_of` expressions that are generated by CBMC in the new SMT backend, we currently have to supply an extra argument `--slice-formula` so that instances of `arrayof_exprt` are removed from the formula to be converted.

> As we move forward with our array-support implementation, we anticipate that the need for this extra flag will be diminished.

Reply:
changed